localstack-core/localstack/services/stepfunctions/asl/eval/environment.py

```diff
@@ -270,7 +270,10 @@ def is_standard_workflow(self) -> bool:
         return self.execution_type == StateMachineType.STANDARD
 
     def is_mocked_mode(self) -> bool:
-        return self.mock_test_case is not None
+        return (
+            self.mock_test_case is not None
+            and self.next_state_name in self.mock_test_case.state_mocked_responses
```

Comment:
thought: interesting that the test case itself is unsufficient and we need to check forthe next state as well. Should we add a comment clarifying in which scenarios this is needed?